packages/flutter_tools/lib/src/android/build_validation.dart

```diff
@@ -17,6 +17,12 @@ void validateBuild(AndroidBuildInfo androidBuildInfo) {
       'For more information see $kSupportedAbis .'
     );
   }
+  if (!buildInfo.mode.isPrecompiled && buildInfo.analyzeAotSize) {
```

Comment:
This validation should go in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/runner/flutter_command.dart#L601

Reply:
Should `final BuildInfo buildInfo = androidBuildInfo.buildInfo;` also be copy and pasted? Seems like we only have access to `BuildMode` in flutter_command.dart which does not have the field `analyzeAotSize`.

Reply:
that is the code where you added analyzeAotSize